java/src/jmri/jmrix/cmri/serial/serialmon/SerialMonFrame.java

```diff
@@ -31,13 +89,466 @@ protected void init() {
+                String logLine = sb.toString();
+                if (!newline.equals("\n")) {
+                    // have to massage the line-ends
+                    int i = 0;
+                    int lim = sb.length();
+                    StringBuffer out = new StringBuffer(sb.length()+10);  // arbitrary guess at space
+                    for ( i = 0; i<lim; i++) {
+                        if (sb.charAt(i) == '\n')
+                            out.append(newline);
+                        else
+                            out.append(sb.charAt(i));
+                    }
+                    logLine = out.toString();
+                }
+                logStream.print(logLine);
+            }
+        }
+    }
+
+    String newline = System.getProperty("line.separator");
+
+    /**
+     * Method to position caret at end of JTextArea ta when
+     * scroll true.
+     * @param ta Reference to JTextArea
+     * @param scroll True to move to end
+     */
+    private void doAutoScroll(final JTextArea ta, final boolean scroll) {
+        SwingUtilities.invokeLater(new Runnable() {
+            @Override
+            public void run() {
+                int len = ta.getText().length();
+                if (scroll) {
+                    ta.setCaretPosition(len);
+                } else if (ta.getCaretPosition()==len && len>0) {
+                    ta.setCaretPosition(len-1);
+                }        
+            }
+        });
+    }
+    /**
+     * Method to test for packet logging
+     * @param OnOff 
+     */
+    public void setMsgLogging( boolean OnOff )
+    {
+        loggingEnabled = OnOff;
+    }    
+    public boolean getMsgLogging()
+    {
+        return loggingEnabled;       
+    }
+    /**
+     * Toggle the packet logging function with one button
+     * @param e 
+     */
+    public synchronized void logButtonActionPerformed(java.awt.event.ActionEvent e) {
+        // start logging by creating the stream
+        if (getMsgLogging()) // Logging enabled, stop logging
+        {
+            if (logStream!=null)
+            {
+                synchronized (logStream)
+                {
+                 logStream.flush();
+                 logStream.close();
+                }   
+           }
+           logStream = null;
+           logMsgButton.setText(Bundle.getMessage("StartLoggingText"));
+           setMsgLogging( false );
+        }
+        else
+        if ( logStream==null)
+        {  // successive clicks don't restart the file
+            // start logging
+            try {
+                logStream = new PrintStream (new FileOutputStream(logFileChooser.getSelectedFile()));
+                logMsgButton.setText(Bundle.getMessage("StopLoggingText"));
+                setMsgLogging( true );
+
+            } catch (Exception ex) {
+                log.error("exception "+ex);
+            }
+        }
+    }
+    
+     /**
+     * Toggle the display on/off
+     * @param e 
```

Comment:
All parameters in Javadoc comments need a description—Jenkins will list param annotations without descriptions as errors in the JavaDocs.

Reply:
Removed @params